### ISSUE TITLE
predict function missing in DBSCAN clustering algorithm module

### ISSUE THREAD

**Issue by b-serra** [elixir-nx/scholar#259]
Is there any reason why DBSCAN module lacks of `predict` function compared to other `clustering` models contained here.

**Comment by b-serra** [comment]
Hi @krstopro! [There is a `fit_predict` method in `scikit-learn`](https://scikit-learn.org/stable/modules/generated/sklearn.cluster.DBSCAN.html#sklearn.cluster.DBSCAN.fit_predict). Anyway I did not notice that `scholar` aims to map `scikit-learn` lib! My bad.

**Comment by krstopro** [comment]
> [There is a fit_predict method in scikit-learn](https://scikit-learn.org/stable/modules/generated/sklearn.cluster.DBSCAN.html#sklearn.cluster.DBSCAN.fit_predict).

Perhaps we should add `fit_predict` here as well. I am not sure about it because `fit_predict` would just return the `labels` of the model fit on `x`.

> Anyway I did not notice that scholar aims to map scikit-learn lib! My bad.

No worries - any suggestions are welcome!
There is a substantial difference how the two actually work because `Scholar` is built on top of `Nx`. In case you are not familiar with it, I suggest having a look at [Introduction to Nx](https://hexdocs.pm/nx/intro-to-nx.html).
